backend/server/api/controllers/todo/controller.js

```diff
@@ -37,6 +38,35 @@ export class Controller {
       console.log(error);
     }
   }
+
+  async search(req, res) {
+    const searchTodo = {
+      useName: req.body.useName,
+      useDescription: req.body.useDescription,
+      useIsDone: req.body.useIsDone,
+      useDueDate: req.body.useDueDate,
+      name: req.body.name,
+      description: req.body.description,
+      isDone: req.body.isDone,
+      dueDate: req.params.dueDate,
+    };
+    // If the request body is empty, the name property will be undefined
+    // If all the searching filters are false, there is nothing to find, so return bad request
+    if (
+      searchTodo.name == undefined ||
+      (!searchTodo.useName &&
+        !searchTodo.useDescription &&
+        !searchTodo.useIsDone &&
+        !searchTodo.useDueDate)
+    )
```

Comment:
Redundant. Just checking for the name would be fine.

Reply:
Well ok, I'll assume that name is a required filter.